src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager.cs

```diff
@@ -145,30 +143,26 @@ internal static void SyncDeferredThemeModeAndResources()
         if(Application.Current == null) return;
 
         ThemeMode themeMode = Application.Current.ThemeMode;
+        var rd = Application.Current.Resources;
 
         bool resyncThemeMode = false;
-        int index = FindLastFluentThemeResourceDictionaryIndex(Application.Current.Resources);
+        int index = LastIndexOfFluentThemeDictionary(rd);
 
-        if(index > 0)
+        if (index == -1 && themeMode != ThemeMode.None)
         {
-            // This means that the devleoper has added the Fluent theme resources manually
-            themeMode = GetThemeModeFromResourceDictionary(Application.Current.Resources);            
+            // This means that ThemeMode was set but Resources were not set during initialization.
+            // Hence we need to resync.
             resyncThemeMode = true;
         }
         else
         {
-            // This means that the developer has set ThemeMode but Resources has not been set
-            // Hence we need to resync the properties but ThemeMode is the one that should be used here.
-            if(themeMode != ThemeMode.None && index != 0)
-            {
-                resyncThemeMode = true;
-            }
-
+            // If index > 0, then Fluent theme dictionary was added manually.
             // If ThemeMode is None, and yet there is a Fluent theme dictionary, hence that was manually set.
-            if(themeMode == ThemeMode.None && index == 0)
+            // Hence we need to resync.
+            if (index > 0 || themeMode == ThemeMode.None)
             {
+                themeMode = GetThemeModeFromSourceUri(rd.MergedDictionaries[index].Source);
```

Comment:
I think this will give "IndexOutOfRangeException" in case of (index == -1 && themeMode == ThemeMode.None) ?

Reply:
Will fix this